### PR TITLE
Cross platform support for window maximizing

### LINKED ISSUE #91: Maximizing window gives error in linux
## Bug
Maximizing window gives error in linux.
```
File "/home/cid0rz/.pyenv/versions/3.11.4/lib/python3.11/tkinter/init.py", line 2276, in wm_state
    return self.tk.call('wm', 'state', self._w, newstate)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
_tkinter.TclError: bad argument "zoomed": must be normal, iconic, or withdrawn
```

### System Information
- OS: arch linux
- Python Version: 3.11

### PR DESCRIPTION
**Pull request by billyeatcookies**
fix #91

### CODE REVIEW

File: biscuit/core/utils/events.py [hunk 1]
```diff
@@ -57,7 +57,10 @@ def quit(self, *_):
         self.base.destroy()
     
     def toggle_maximize(self, *_):
-        self.base.wm_state('normal' if self.maximized else 'zoomed')
+        if self.base.sysinfo.os == "Windows":
```
Comment (rdbende) [inline biscuit/core/utils/events.py:60]:
```suggestion
        if self.base.sysinfo.os != "Linux":
```

This way it supports macOS as well.

**Comment by billyeatcookies** [comment]
thats a really nice fix @rdbende ❤ i was not thinking about mac at all :)

**Comment by rdbende** [comment]
I mean, I hope that the actual value is "Linux" and not "linux" or something else. I just blindly commented the suggestion, and not tested, or looked at the source code :D

**Comment by billyeatcookies** [comment]
Yess it's `Linux` ! base.sysinfo simply uses [`platform.system`](https://docs.python.org/3/library/platform.html#platform.system), values can be  Linux, Darwin, Java, Windows, etc.

The idea of keeping a reference to the `base` component was a solution for widgets to communicate between each other easily, though I'm still not sure if its actually good or there are better ways to handle it 😅

**Comment by rdbende** [comment]
For things like thins I would actually prefer using plain `platform.system`, `sys.platform` or some constant like `RUNNING_ON_WINDOWS`, because then I know what the value will be. There are so many different ways to detect the current platform, that it's sometimes hard to guess what the actual value will be.

**Pull request merged**